tenseal/tensealcontext.h

```diff
@@ -176,6 +176,23 @@ class TenSEALContext {
         return encoder_factory->get<T>();
     }
 
+    /*
+    Template encoding functions to choose between the use of BatchEncoder or
+    CKKSEncoder.
+    */
+    template <class BatchEncoder>
+    void encode(vector<int64_t>& vec, Plaintext& pt) {
+        auto encoder = this->get_encoder<BatchEncoder>();
+        encoder->encode(vec, pt);
+    }
+
+    template <class CKKSEncoder>
+    void encode(vector<double>& vec, Plaintext& pt, double scale = 0) {
+        if (scale == 0) scale = pow(2, 40);  // TODO: get from TenSEALContext
+        auto encoder = this->get_encoder<CKKSEncoder>();
+        encoder->encode(vec, scale, pt);
+    }
+
```

Comment:
This looks neat, great job!

Just a few suggestions:
- You could the template support for IntegerEncoder too, even if we drop bfvnaivevector, just for covering the whole interface.
- You can add a default template<> implementation that throws an exception, for making sure that other types won't be used.
- I think it works to add a decode method too, for symmetry.
- The tensor implementations could use directly the encode/decode methods, instead of loading the encoder via get.
- For a robust interface, maybe the actual implementation for encode/decode should be placed in the TenSEALEncoder class, while TenSEALContext only forwards the parameters to the encoder. This way, the context class stays lighter. 
- If you expose the encode/decode methods, you can drop the get call to the TenSEALEncoder too.

Reply:
We should certainly work on this, but I would prefer to let it for another PR.

Reply:
I think #50 refer to this (just to link to issue to where it first introduced).